File: pa5/DictionaryTest.c
// Aaron Chan
// 5/28/14
// DictionaryTest.c
// pa5

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
  Dictionary A = newDictionary();
  //   char* k;
  //   char* v;
  char* word1[] = {"this","is","quite", "confusing"};
  char* word2[] = {"one","two","three", "four"};
  int i;

  for(i=0; i<4; i++){
    insert(A, word1[i], word2[i]);
  }

  printDictionary(stdout, A);

  for(i=0; i<4; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }

  delete(A, "one");
  delete(A, "three");

  printDictionary(stdout, A);

  for(i=0; i<4; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    makeEmpty(A);
    printf("%s\n", (isEmpty(A)?"true":"false"));

    freeDictionary(&A);

    return(EXIT_SUCCESS);
  }

File: pa3/DictionaryTest.java
// Aaron Chan - aahchan
// 4/27/16
// pa3
// DictionaryTest.java
// To serve as a test client for the Dictionary ADT while it is under construction

class DictionaryTest{
    public static void main (String[] args){
	Dictionary Test = new Dictionary();
	System.out.println(Test.isEmpty());
	Test.insert("4","f");
	System.out.println(Test.isEmpty());
	Test.insert("5","d");
	System.out.println(Test.size());
	Test.insert("6","3");
	Test.insert("7","c");
	Test.insert("3","e");
	Test.insert("9","3");
	//Test.delete("4");
	//Test.insert("6","e"); //throws exception 
	System.out.println(Test);
	System.out.println(Test.lookup("7"));
	System.out.println("********");
	Test.delete("6");
	System.out.println(Test.size());
	System.out.println(Test.lookup("6"));
	Test.delete("5");
	Test.delete("4");
	System.out.println(Test.lookup("9"));
	System.out.println(Test.size());
	System.out.println(Test);
	Test.makeEmpty();
	System.out.println(Test.size() + " " + Test.isEmpty());
    
    }
}
